core/src/main/java/org/jruby/RubyBasicObject.java

```diff
@@ -802,19 +805,25 @@ public RubyString convertToString() {
      * format of a class name and a hex string inside of #<>.
      */
     @Override
-    public RubyString anyToString() {
-        String cname = getMetaClass().getRealClass().getName();
-        String hex = Integer.toHexString(System.identityHashCode(this));
+    public IRubyObject anyToString() {
+        Ruby runtime = getRuntime();
+
         /* 6:tags 16:addr 1:eos */
-        ByteList bytes = new ByteList(2 + cname.length() + 3 + hex.length() + 1);
-        bytes.append('#').append('<');
-        bytes.append(cname.getBytes(RubyEncoding.UTF8));
-        bytes.append(':').append('0').append('x');
-        bytes.append(hex.getBytes());
-        bytes.append('>');
-        RubyString str = RubyString.newString(getRuntime(), bytes, UTF8Encoding.INSTANCE);
-        str.setTaint(isTaint());
-        return str;
+        RubyString bytes = runtime.newString("#<");
```

Comment:
just a q: so the direct `ByteList` `concat` wasn't good here, does it get problematic for any reason? 
isn't the new way going to be doing a lot of "unnecessary" Java `String`-> `byte[]` conversion?

Reply:
That is a good question I don't recall offhand.  It is possible there was no issue but I learned that these methods are not quite a sensible as you may think.  I will try it out quickly.

Reply:
So I looked at this and I think I can reduce the cost substantially and use a bytelist but there is a specific reason why it works.  First when we have m17n data getting combined into a string we cannot use bytelist since bytelist has no idea how to combine two different encoded things together.  In this very particular case we only have 7bit ascii strings and something else.  So I think I can just mark the new string as same encoding as the something else and it will work....almost anyways.  If the encoding used is not isAsciicompat we will generate garbage.  OTOH my current code was broken in the same way so we are no worse off.

Reply:
Addendum since I am thinking about this.  I should try the last broken part here.  I suspect the types() method in RubyStringBuilder is actually what the non-ascii name will display like.   So I need to try that.